### PR TITLE
Create setup script

### LINKED ISSUE #158: Make startup script
Create a startup script for new members. This will initialize their local db with the auth permission to give them all the admin & members access

### PR DESCRIPTION
**Pull request by ayush-goyal**
### Description
- Create an interactive setup script for new members to get setup in the api repo
- Checks that they have the proper `.env` file and google cloud secrets
- Ensures the databases are running locally and that migrations are run
- Ensures that their permission and profile is setup locally

Sample screenshot from the script execution
Screenshot 2023-08-04 at 9 58 04 PM

### Issues

- Closes #158

### CODE REVIEW

File: scripts/src/setup.ts [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  const mongoClient = new MongoClient("mongodb://localhost");
+  try {
+    await mongoClient.connect();
+  } catch (error: any) {
+    console.log(
+      "Could not connect to MongoDB. Please ensure your connection string in .env has the correct user and password. " +
+        error.message
+    );
+  }
+
```
Comment (thaneesh-babu) [inline scripts/src/setup.ts:138]: can we log a successful mongodb connection message here before checking the postgres setup

**Review by thaneesh-babu — COMMENTED** [review]
Great work! Commented a small thing and would like @Roshan-D to look at this as well since my postgres setup is still iffy

**Review by thaneesh-babu — CHANGES_REQUESTED** [review]
Nvm got postgres working. So it worked well but I had a few other suggestions:
- If you select no for any of the inputs it still continues so I'd guess we wanna exit the script in that case
- For some of the questions like enter uid, name etc, it would be better if we loop for a valid input in case of input invalidation just so that we don't have to restart the script. Something like this:

```
let userId = "";
  while (1) {
    userId = await input({
      message: chalk.cyan("What is your ") + chalk.gray("uid") + chalk.cyan("?: "),
    });
    if (userId.length < 10) {
      logError("Invalid uid. Please ensure you have copied the correct uid.");
      continue;
    }
    break;
  }
```

Everything else is dope!

**Comment by ayush-goyal** [comment]
Fixed @thaneesh-babu

**Review by thaneesh-babu — APPROVED** [review]
approved if u can just resolve conflicts and then merge

**Comment by ayush-goyal** [comment]
@Roshan-D if u wanna take a look too lmk

**Review by Roshan-D — APPROVED** [review]
this is amazing! love it!

**Pull request merged**